### DISCUSSION TITLE
How to change xaxis labels styling on column chart?

### DISCUSSION THREAD

**Discussion by genialkartik** [apexcharts/apexcharts.js#2913] [Q&A]
How to change x-axis labels background-colors, border's width, and radius?

I'm using this configuration:

```js
 const options = {
   ...
    xaxis: {
      categories: props.dataX,
      labels: {
        style: {
          fontSize: "14px",
          colors: "red",
          backgroundColor: "#ff0000",
          cssClass: "chart-label-x",
        },
      },
      axisTicks: { show: false },
    },
    ...
    },
  };
```

### Current Chart Design:

<img width="833" alt="Screenshot 2022-01-24 at 10 23 02 AM" src="https://user-images.githubusercontent.com/32240906/150724076-e0ff159f-13e4-416a-8121-7fae9a404eda.png">

### Targeted Chart Design:

<img width="756" alt="Screenshot 2022-01-24 at 10 23 37 AM" src="https://user-images.githubusercontent.com/32240906/150724125-3629ae6c-6dd4-40bb-966d-f6c94762a2e9.png">

Language stacks using:
1. React.js
2. Material-UI

Note:
Adding external css class (`cssClass: "chart-label-x"`) and defining properties in external css style file (styles.css) didn't help.

```
.chart-label-x {
  border-style: solid;
  border: 5px solid red;
  background: blue;
  color: yellow;
}
```

**Comment by genialkartik** [comment]
Any update on this? 
Anyone?

**Comment by hossamrq** [comment]
Any update please on this @genialkartik 🙏🏻

**Comment by gwyn17** [comment]
One crude way to do this is to get all the .apexcharts-xaxis-label and create a `<rect>` that overlaps them.  The following pure-JS works for me.  YMMV. 
```
        // Run this after you call chart.render();

        var parent = this.chart.w.globals.dom.baseEl;
        var labels = parent.querySelectorAll( ".apexcharts-xaxis-label" );
        parent = parent.querySelector( ".apexcharts-svg" );
        var box = parent.getBoundingClientRect();
        var px = 10;    // padding
        var py = 5;
        var dx = box.x+px;  // offset
        var dy = box.y+py;
        px *= 2;        // efficiency
        py *= 2;
        labels.forEach( function ( el, i, list ) {
                var box = el.getBoundingClientRect();
                var rect = document.createElementNS( 'http://www.w3.org/2000/svg', 'rect' );
                rect.setAttribute( "class", "apexcharts-xaxis-box" );
                rect.setAttribute( 'x', Math.round( box.x - dx ) );
                rect.setAttribute( 'y', Math.round( box.y - dy ) );
                rect.setAttribute( 'width', Math.round( box.width + px ) );
                rect.setAttribute( 'height', Math.round( box.height + py ) );
                rect.setAttribute( 'rx', 3 );
                parent.appendChild( rect );
            } );
```

These `<rect>s` are styled thusly : 
```
.apexcharts-xaxis-box {
    fill: transparent;
    stroke: blue;
}
```

The above is only a proof of concept.  Each time you call `chart.render()` you might have to move your `<rect>s` around.

You are going to have to play around with Z order to get the text on background you show in your example.
